src/backend/core/product/serializers.py

```diff
@@ -508,3 +506,20 @@ class Meta:
             "slug",
             "product_variants",
         )
+
+
+class ProductStorefrontSerializer(TranslatedSerializerMixin, ModelSerializer):
+    """
+    Product serializer used for storefront.
+    Only one translation is returned (see TranslatedSerializerMixin)
+    """
+
+    class Meta:
+        model = Product
+        fields = (
```

Comment:
It'd be greate to also return primary photo and price (request should of course contain pricelist ID) 😊

Reply:
Primary image added.

Actually, there's a problem with prices, since we have prices related to product variant, not product.
So, we need to somehow select the *default* variant.

@michalpulpan what do you think about it?

Reply:
Good point, I didn't think about that. 

> So, we need to somehow select the default variant.

That would be an option, or we can do something like "price aggregation" on `Product` and calculate min-max price for give pricelist?

Reply:
or we can simply return all variants with prices, which might be the best solution.
And then when developing storefront, we select on the frontend what we want to display...

Reply:
> or we can simply return all variants with prices, which might be the best solution.
> And then when developing storefront, we select on the frontend what we want to display...

Yes, I'm for it 😊. It wont be much data if we paginate the data and limit each page to ~32 results.